[O11y][Nagios_xi] Enable secret for the api_key field, update format version and add dashboard-level filters (#9996)
elastic/integrations: packages/nagios_xi/_dev/build/docs/README.md

# Nagios XI

## Overview

The Nagios XI integration is used to fetch observability data from [Nagios XI](https://www.nagios.org/documentation/) and ingest it into Elasticsearch.

Use the Nagios XI integration to:

> - Collect metrics related to the current load, current users, ping, http, ssh, root partition, swap users, total processes, round-trip time, and packet loss. Additionally, gather logs related to system events.

Reviewer: ```suggestion
- Collect metrics on current load, users, ping, HTTP, SSH, root partition, swap users, total processes, round-trip time, and packet loss, along with system event logs.
```

> - Create visualizations to monitor, measure and analyze the usage trend and key data, and derive business insights.

Reviewer: ```suggestion
- Create visualizations to monitor, measure, and analyze usage trends and key data for business insights.
```

- Set up alerts to minimize Mean Time to Detect (MTTD) and Mean Time to Resolve (MTTR) by quickly referencing relevant logs during troubleshooting.

## Data streams

The Nagios XI integration collects logs and metrics data.

Logs provide insights into operations and events within the Nagios XI environment. The `Log` data stream collected by the Nagios XI integration is events. This allows users to track `system events`, understand their causes, and address issues related to infrastructure monitoring and alert management.

> Metrics offer insights into the performance and health of user's Nagios XI instance. The `Metric` data stream collected by the Nagios XI integration are host and service. These enable users to monitor and troubleshoot the performance of hosts and services within their Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and various other critical indicators.

Reviewer 2: ```suggestion
Metrics provide insights into the performance and health of your Nagios XI instance. The Nagios XI integration collects `host` and `service` metric data streams. These metrics enable you to monitor and troubleshoot the performance of hosts and services within your Nagios XI environment, covering aspects such as `network round trip time`, `packet loss`, `service load`, `user count`, and other critical indicators. 
```
Change `users` to `yours` for a more direct address to the reader. We follow the same practice for other packages as well.

Data streams:

> - `events`: This data stream gives Nagios XI system event logs.

Reviewer: ```suggestion
- `events`: Provides Nagios XI system event logs.
```

> - `host`: This data stream gives Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.

Reviewer: ```suggestion
- `host`: Provides Nagios XI Host Round Trip Travel Time (rta) and Packet Loss (pl) metrics.
```

> - `service `: This dataset gives Nagios XI services current load, current users, ping, http, ssh, root partition, swap users and total processes metrics by default.

Reviewer: ```suggestion
- `service `: Provides Nagios XI service metrics by default, including current load, current users, ping, HTTP, SSH, root partition, swap users, and total processes.
```

Note:

> - Users can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.

Reviewer 2: ```suggestion
You can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.
```

> - Users can monitor and see the log inside the ingested documents for Nagios XI in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.

Reviewer: ```suggestion
- Users can monitor and view logs from the ingested documents for Nagios XI in the `logs-*` index pattern in `Discover`. For metrics, the index pattern is `metrics-*`.
```

## Compatibility

This integration has been tested against `Nagios-XI Version: 5.8.7`

## Prerequisites

> User need Elasticsearch for storing and searching user's data and Kibana for visualizing and managing it. User can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended or self-manage the Elastic Stack on user's own hardware.

Reviewer: ```suggestion
Users need Elasticsearch for storing and searching their data and Kibana for visualizing and managing it. They can either use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on their own hardware.
```

In order to ingest data from Nagios XI:

> - User must know the host for Nagios XI, add that host while configuring the integration package.

Reviewer: ```suggestion
- Users must know the host for Nagios XI and add that host when configuring the integration package.
```

## Setup

For step-by-step instructions on how to set up an integration, see the [Getting started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide.

## Validation

After the integration is successfully configured, clicking on the Assets tab of the Nagios XI Integration should display a list of available dashboards. Click on the dashboard available for user's configured data stream. It should be populated with the required data.

> ## Logs reference

Reviewer 2: Why this change? Is it in accordance to any guideline, or any other example where we may be doing the same?
